.github/workflows/visual_tests.yml

```diff
@@ -1,8 +1,13 @@
 name: Visual Tests
 
+concurrency:
+  group: wf-${{github.repository}}-${{github.event.pull_request.number}}-${{github.workflow}}
+  cancel-in-progress: true
+
 on:
   push:
   pull_request:
+    types: [labeled, opened, synchronize, reopened]

```

Comment:
I think that it is not necessary to rerun tests on label. Usually label is added on PR creating. And It is possible to rerun tests after adding label. Also, I think that 'force visual tests' is not good name for label. It will be better 'check all tests' or similar. It is no good to rerun tests after adding label 21_2 for example.

Reply:
Agree with removing the `labeled` event type, moreover, Eugene has the same concerns. As for the label name, I think, we can discuss it in the further PRs